task_processing/plugins/mesos/execution_framework.py

```diff
@@ -527,20 +572,37 @@ def resourceOffers(self, driver, offers):
 
             accepted.append('offer: {} agent: {} tasks: {}'.format(
                 offer.id.value, offer.agent_id.value, len(tasks_to_launch)))
-            driver.launchTasks(offer.id, tasks_to_launch)
 
+            task_launch_failed = False
+            try:
+                driver.launchTasks(offer.id, tasks_to_launch)
+            except (socket.timeout, Exception):
+                log.warning('Failed to launch following tasks {tasks}.'
+                            'Thus, moving them to UNKNOWN state'.format(
+                                tasks=', '.join([
+                                    task.task_id.value for task in
+                                    tasks_to_launch
+                                ]),
+                            )
+                            )
+                task_launch_failed = True
+
+            current_task_state = 'UNKNOWN' if task_launch_failed else \
```

Comment:
mesos_status_to_event uses TASK_UNKNOWN to map mesos state to an event.

Reply:
This is different than TASK_UNKNOWN. We set the state of tasks to 'UNKNOWN' that are not launched successfully. When we reconcile the state of these tasks, mesos will return TASK_LOST.

Reply:
You are right. This UNKNOWN is not put to executor's queue. It would be helpful to add a comment on this.